.github/workflows/publish.yml

```
@@ -11,23 +11,38 @@ on:
     paths:
       - "src/**"
       - ".github/workflows/publish.yml"
-  repository_dispatch:
-    types: [create-release]
-    client_payload:
+  workflow_dispatch:
+    inputs:
       release_tag:
         required: true
+        description: 'Tag for the new release'
+    prerelease:
+      description: 'If true, the release will be set as a prerelease and not impact the latest tag'
+      type: boolean
+      required: true
+      default: false
+
 
 jobs:
   create_release:
-    if: github.event_name == 'repository_dispatch'
+    if: github.event_name == 'workflow_dispatch' && github.ref == 'refs/heads/main'
```

how about
```suggestion
    if: github.event_name == 'workflow_dispatch' && (github.ref == 'refs/heads/main' || github.events.input.prerelease == 'true')
```

seems fine to create prereleases even when it's not main
(`== 'true'` is needed because all GitHub Actions inputs are ultimately just strings)

---

Yea you're right this will help with testing too